server/service/apple_mdm.go

```diff
@@ -1749,6 +1749,30 @@ func (svc *Service) BatchSetMDMAppleProfiles(ctx context.Context, tmID *uint, tm
 		profs = append(profs, mdmProf)
 	}
 
+	allProfs, _, err := svc.ds.ListMDMConfigProfiles(ctx, tmID, fleet.ListOptions{PerPage: 0})
```

Comment:
@roperzh, these are the changes that could impact the puppet flow, which still uses the legacy service [method](https://github.com/fleetdm/fleet/blob/22387ba8a4035cad08ba68a267e3e57bea116edf/ee/server/service/mdm.go#L902-L903). It would be great to get your input on places where you think we could beef up test coverage/manual QA for this change.

Reply:
I think this is fine, I'm a bit concerned about adding an extra db call on each puppet run

Reply:
Yeah, that was my worry too.

Reply:
I just checked and this is going to be called with a context that forces reading from the main instance. Any thoughts on skipping this check for the puppet endpoint? knowing that it currently can only be used for macOS profiles

Reply:
> currently can only be used for macOS profiles
Does that exclude macOS declarations?

The concern would be that Puppet tries to create a macOS profile that has the same name as an existing Windows profile or macOS declaration.

Reply:
yes, but given the practical circumstances of how the module is used, I'm vouching to tackle the biggest concern regarding DB load instead.

Reply:
I'm ok with that. Do you want to add a commit with the change?

Reply:
I think it would be just receiving the param, we can trust the integration tests.

Reply:
So if `skipBulkPending` is `true`, we skip the cross-platform name check too?

Reply:
that makes sense! feel free to add another param if that makes more sense, or rename the variable, or just add the `if`!